pkg/scheduler/statefulset/autoscaler.go

```diff
@@ -103,7 +103,6 @@ func (a *autoscaler) Start(ctx context.Context) {
 }
 
 func (a *autoscaler) Autoscale(ctx context.Context, attemptScaleDown bool, pending int32) {
-	a.trigger <- pending
```

Comment:
would be worth removing `syncAutoscale` and just do the `syncAutoscale` logic in this `Autoscale` method?